dbt/adapters/athena/python_submissions.py

```diff
@@ -0,0 +1,123 @@
+import time
+from functools import cached_property
+from typing import Any, Dict
+
+import boto3
+
+from dbt.adapters.athena.connections import AthenaCredentials
+from dbt.adapters.base import PythonJobHelper
+from dbt.events import AdapterLogger
+from dbt.exceptions import DbtRuntimeError
+
+DEFAULT_POLLING_INTERVAL = 2
+SUBMISSION_LANGUAGE = "python"
+DEFAULT_TIMEOUT = 60 * 60 * 24
```

Comment:
You can configure it for an individual model with `self.parsed_model["config"].get("timeout", DEFAULT_TIMEOUT)`. Do you think it would better to also have global option? I didn't want to make the profiles.yml too complicated.

Reply:
I believe that having global is better and cleaner.

Reply:
ok I will try to add the global config for this

Reply:
If it's too complicated we can consider a future iteration

Reply:
I tried this but my implementation lacked some polish. I will tackle this in an update.